### PR TITLE
Create workflow for building and publishing docker images

### PR DESCRIPTION
**Pull request by lasseheia**

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -3,7 +3,8 @@ updates:
   - package-ecosystem: "pip"
     directory: "/"
     schedule:
-      interval: "daily"
+      interval: "weekly"
+    target-branch: "develop"
```
Comment (sixcare) [inline .github/dependabot.yml:7]:
```suggestion
    target-branch: "main"
```

Comment (sixcare) [reply .github/dependabot.yml:7]: just test

File: .github/workflows/publish-docker-image.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Create and publish a Docker image
+
+on:
```
Comment (sixcare) [inline .github/workflows/publish-docker-image.yaml:3]:
```suggestion
on:
  workflow_dispatch:
```

**Review by sixcare — CHANGES_REQUESTED** [review]
Change image in docker compose.

Change dev docker image or remove?

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,23 @@
-FROM python:3.11-bullseye
-
-RUN pip install --upgrade pip
+# Start with a smaller base image like python:slim-buster
```
Comment (lasseheia) [inline Dockerfile:1]:
```suggestion
```

File: Dockerfile [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
-
-HEALTHCHECK --interval=30s --timeout=5s --start-period=20s --retries=5 CMD [ "curl", "-fs", "http://localhost:5000/test"]
+# Install build dependencies, then run `pip install`, then remove unneeded build dependencies all in a single step.
+# Combined RUN statements into one to reduce number of image layers.
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends gcc postgresql-client libpq-dev libc6-dev && \
+    pip install --upgrade pip && \
+    pip install --no-cache-dir .['prod'] && \
+    apt-get purge -y --auto-remove gcc libpq-dev libc6-dev && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (lasseheia) [inline Dockerfile:18]:
```suggestion

HEALTHCHECK --interval=30s --timeout=5s --start-period=20s --retries=5 CMD [ "curl", "-fs", "http://localhost:5000/test"]

```

File: Dockerfile [hunk 5]
```diff
@@ -1,15 +1,24 @@
-FROM python:3.11-bullseye
-
-RUN pip install --upgrade pip
+FROM python:3.11-slim-bullseye
 
+# Set work directory
 WORKDIR /turplanlegger
 
+# Copy rest of the application code
```
Comment (lasseheia) [inline Dockerfile:6]:
```suggestion
```

**Review by sixcare — APPROVED** [review]
gj

**Pull request merged**